python/src/numerous/collection/collection.py

```diff
@@ -0,0 +1,16 @@
+"""Get or create a collection by name."""
+
+from typing import Optional
+
+from numerous._client import Client, _open_client
+from numerous.collection.numerous_collection import NumerousCollection
+
+
+def collection(
+    collection_name: str, numerous_open_client: Optional[Client]
```

Comment:
1. Maybe the client argument could be `_client: Optional[Client] = None` instead. The underscore indicates that it is internal functionality to change the client, and the default value ensures that users can call it with just `collection("some name")
2. Shouldn't the first argument be a "key"? And it is redundant to write `collection_key`, just call the argument `key`.